common/junit-platform-native/src/main/java/org/graalvm/junit/platform/JUnitPlatformFeature.java

```diff
@@ -143,14 +166,18 @@ private TestPlan discoverTestsAndRegisterTestClassesForReflection(Launcher launc
                 .map(ClassSource.class::cast)
                 .map(ClassSource::getJavaClass)
                 .forEach(this::registerTestClassForReflection);
-
-        return testPlan;
     }
 
     private void registerTestClassForReflection(Class<?> clazz) {
         debug("Registering test class for reflection: %s", clazz.getName());
         nativeImageConfigImpl.registerAllClassMembersForReflection(clazz);
```

Comment:
I think interfaces and superclasses of member classes also need to be registered for reflection to support scenarios such as this:

```java
abstract BaseTests {
    @Test test() {}
}

@ParameterizedClass
class OuterTests {
	@Nested class InnerTests extends BaseTests {}
}
```

Reply:
I extracted [this code](https://github.com/graalvm/native-build-tools/pull/693/files#diff-88ba6393b3a8379912730935514a00f87cbf552725c49effd00eeb8b3f17cbd0R176) from `registerAllClassMembersForReflection` function. That way we are registering all class members + interfaces + superclass from inner classes.

@marcphilipp Could you please add the above test [here](https://github.com/graalvm/native-build-tools/tree/master/common/junit-platform-native/src/test/java/org/graalvm/junit/jupiter) once you release next JUnit version that supports `@ParameterizedClass`?

Reply:
> I extracted [this code](https://github.com/graalvm/native-build-tools/pull/693/files#diff-88ba6393b3a8379912730935514a00f87cbf552725c49effd00eeb8b3f17cbd0R176) from `registerAllClassMembersForReflection` function. That way we are registering all class members + interfaces + superclass from inner classes.

LGTM

> Could you please add the above test [here](https://github.com/graalvm/native-build-tools/tree/master/common/junit-platform-native/src/test/java/org/graalvm/junit/jupiter?rgh-link-date=2025-04-28T10%3A44%3A01Z) once you release next JUnit version that supports `@ParameterizedClass`?

Would it be okay to use a milestone version (5.13.0-M2) only for the tests in `junit-platform-native`? If so, we could add the test right now. WDYT?

Reply:
This will require another issue to be opened, and should not be mixed into this PR. Please open an issue with this request (e.g. something like "Add testing of JUnit support for JUnit milestone releases")

Reply:
I submitted https://github.com/graalvm/native-build-tools/pull/722 as a draft. I'll update it once 5.13.0 is released.